android-lib/src/main/java/com/kinvey/android/async/AsyncBatchPushRequest.kt

```diff
@@ -130,7 +130,7 @@ class AsyncBatchPushRequest<T : GenericJson>(
                         manager.deleteCachedItem(syncItem[_ID] as String?)
                     }
                 } catch (e: IOException) {
-                    val err = KinveyUpdateSingleItemError(e, item)
+                    val err = KinveyUpdateSingleItemError(e, item as GenericJson)
```

Comment:
since this PR seems to be part of the other one, maybe better to merge #258 first so we don't have to repeat the same comments here